pkg/trace/propagation.go

```diff
@@ -30,14 +46,26 @@ func (t *tracer) fromHeaders(ctx context.Context, hdrs map[string][]string, name
 }
 
 // toHeaders writes the current trace context into a headers map
-func (t *tracer) toHeaders(ctx context.Context) map[string][]string {
+func (t *honeycombTracer) toHeaders(ctx context.Context) map[string][]string {
 	result := http.Header{}
 
-	if t.Honeycomb.Enabled {
-		if span := trace.GetSpanFromContext(ctx); span != nil {
-			result.Set(propagation.TracePropagationHTTPHeader, span.SerializeHeaders())
-		}
+	if span := trace.GetSpanFromContext(ctx); span != nil {
+		result.Set(propagation.TracePropagationHTTPHeader, span.SerializeHeaders())
 	}
 
 	return result
 }
+
+func (t *otelTracer) newTransport(old http.RoundTripper) http.RoundTripper {
+	return otelhttp.NewTransport(old)
+}
+
+// fromHeaders fetches trace info from a headers map
+func (t *otelTracer) fromHeaders(ctx context.Context, hdrs map[string][]string, name string) context.Context {
+	return ctx
+}
+
+// toHeaders writes the current trace context into a headers map
+func (t *otelTracer) toHeaders(ctx context.Context) map[string][]string {
+	return http.Header{}
+}
```

Comment:
I think I know why these are blank (this is handled automatically by the round-tripper, right?), but it might help to add a comment about it.

Reply:
This was an unfortunate byproduct of putting OTel and Honeycomb under the same interface.

As you mentioned in another comment, OTel introduces really great support for auto instrumentation that would take care of 80% of what we generate by hand. This instrumentation also handles propagation for us and completely gets rid of the need for manual header handling.

I am going to look at moving towards that approach in another PR. My idea is to replace the manual handling, for both OTel and Honeycomb, with a wrapper similar to the automatic OTel instrumentation. You can see an example of the automatic instrumentation here https://github.com/getoutreach/gobox/pull/167/files#diff-d36e35946980b42d4d9b45ea6567e51f1efb2d63bb5078c89abc2b7f83c36e4aR181.